src/ng/directive/input.js

```diff
@@ -2142,7 +2142,7 @@ var ngValueDirective = function() {
    *  makes it possible to use ngValue as a sort of one-way bind.
    */
   function updateElementValue(element, attr, value) {
-    element.prop('value', value);
+    element.prop('value', angular.isUndefined(value) ? null : value);
```

Comment:
Another idea (to avoid the arbitrary conversion to `null`) is to change to `element[0].value = value`. (I am not sure what is better.) It will not matter in most cases (e.g. when used with native inputs), only if someone is using it with a custom input.

##
It turns out that in IE9 `input.value = null` results in `input.value === 'null'` (same for `undefined`). So we have a problem there.

It matters in cases where `value` in always `undefined` (so `updateElementValue` will only run once at the beginning). Previously, `element.prop('value', undefined)` would be ignored (and the initial `value` will be kept), but now it will be set to `'null'` (or `'undefined'`).

We could check for IE9 and set the value to `''` instead (and also add a "known issue" in the docs):

```js
var propValue = angular.isDefined(value) ? value : (msie === 9) ? '' : null;
element.prop('value', propValue);
```

(It is not ideal, but it is a good middleground imo - plus it will only affect IE9 users.)